openarm/damiao/__main__.py

```diff
@@ -1,231 +1,519 @@
+        # Protection Limits
+        "over_temperature": motor.set_over_temperature,
+        "over_current": motor.set_over_current,
+        # Mapping Limits
+        "position_limit": motor.set_position_limit,
+        "velocity_limit": motor.set_velocity_limit,
+        "torque_limit": motor.set_torque_limit,
+        # Control Loop Parameters
+        "velocity_kp": motor.set_velocity_kp,
+        "velocity_ki": motor.set_velocity_ki,
+        "position_kp": motor.set_position_kp,
+        "position_ki": motor.set_position_ki,
+        # Current and Speed Loop Parameters (NEW)
+        "current_loop_bandwidth": motor.set_current_loop_bandwidth,
+        "speed_loop_damping": motor.set_speed_loop_damping,
+        "speed_loop_filter_bandwidth": motor.set_speed_loop_filter_bandwidth,
+        "current_loop_gain": motor.set_current_loop_gain,
+        "speed_loop_gain": motor.set_speed_loop_gain,
+        # Motion Parameters
+        "acceleration": motor.set_acceleration,
+        "deceleration": motor.set_deceleration,
+        "max_speed": motor.set_max_speed,
+        # Communication Parameters
+        "master_id": motor.set_master_id,
+        "slave_id": motor.set_slave_id,
+        "timeout": motor.set_timeout,
+        "can_baudrate": motor.set_can_baudrate,
+        # Note: Read-only parameters are not included in setter methods
+    }
+
+    if param_name not in param_methods:
+        _error(f"Unknown parameter: {param_name}")
+        _error(f"Available parameters: {', '.join(param_methods.keys())}")
+        sys.exit(1)
+
+    # Handle int vs float parameters
+    if param_name in ["master_id", "slave_id", "timeout", "can_baudrate"]:
+        value = int(value)
+    else:
+        value = float(value)
+
+    result = await param_methods[param_name](value)
+    _output(f"{param_name} set to: {result}")
+
+
+def _run_async(coro: Coroutine[Any, Any, None]) -> None:
+    """Run async coroutine in CLI context."""
+    asyncio.run(coro)
 
 
 def _main() -> None:
-    parser = argparse.ArgumentParser()
+    """Execute main CLI entry point."""
+    parser = argparse.ArgumentParser(
+        description="Damiao motor control CLI",
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
     subparsers = parser.add_subparsers(dest="command", required=True)
 
+    # Common arguments
+    def add_common_args(parser: argparse.ArgumentParser) -> None:
+        parser.add_argument("--iface", default="can0", help="CAN interface to use")
+        parser.add_argument(
+            "slave_id", type=int, help="Motor slave ID (for sending commands)"
+        )
+        parser.add_argument(
+            "master_id", type=int, help="Motor master ID (for receiving responses)"
+        )
+
+    def add_motor_type_arg(parser: argparse.ArgumentParser) -> None:
+        parser.add_argument(
+            "--motor-type",
+            required=True,
+            choices=[t.value for t in MotorType],
+            help="Motor type",
+        )
```

Comment:
since `add_motor_type_arg` always come after `add_common_args`, i think we can just merge them:

```suggestion
    # Common arguments
    def add_common_args(parser: argparse.ArgumentParser) -> None:
        parser.add_argument("--iface", default="can0", help="CAN interface to use")
        parser.add_argument(
            "slave_id", type=int, help="Motor slave ID (for sending commands)"
        )
        parser.add_argument(
            "master_id", type=int, help="Motor master ID (for receiving responses)"
        )
        parser.add_argument(
            "--motor-type",
            required=True,
            choices=[t.value for t in MotorType],
            help="Motor type",
        )
```